### ISSUE TITLE
[mfront] Warnings in behavior

### ISSUE THREAD

**Issue by MorganeSteins** [thelfer/tfel#257]
Warnings appear while using Mfront in MANTA. Here are the outputs of failed examples due to these warnings. They pinpoint the faulty lines of code.

1. Unused variable in Kirschoff-SaintVenant :
```
/builds/manta-project/manta/build_pumi/examples/charpy/generic/src/SaintVenantKirchhoffElasticity-generic.cxx:377:29: error: unused variable 's0' [-Werror,-Wunused-variable]
tfel::math::stensor<1,real> s0;
                            ^
/builds/manta-project/manta/build_pumi/examples/charpy/generic/src/SaintVenantKirchhoffElasticity-generic.cxx:536:29: error: unused variable 's0' [-Werror,-Wunused-variable]
tfel::math::stensor<2,real> s0;
```

2. Capture not required in lambda function in `PhaseFieldDisplacementSpectralSplit`
I suspect that we copy-pasted some code here, although I do not know how it was done by the MANTA developers.
```
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:29:19: error: lambda capture 'emin' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
const auto df = [&emin](const real x)
                  ^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:115:1: note: in instantiation of member function 'tfel::material::PhaseFieldDisplacementSpectralSplit<tfel::material::ModellingHypothesis::AXISYMMETRICALGENERALISEDPLANESTRAIN, double, false>::integrate' requested here
~PhaseFieldDisplacementSpectralSplit()
^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:29:19: error: lambda capture 'emin' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
const auto df = [&emin](const real x)
                  ^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:115:1: note: in instantiation of member function 'tfel::material::PhaseFieldDisplacementSpectralSplit<tfel::material::ModellingHypothesis::AXISYMMETRICAL, double, false>::integrate' requested here
~PhaseFieldDisplacementSpectralSplit()
^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:29:19: error: lambda capture 'emin' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
const auto df = [&emin](const real x)
                  ^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:115:1: note: in instantiation of member function 'tfel::material::PhaseFieldDisplacementSpectralSplit<tfel::material::ModellingHypothesis::PLANESTRAIN, double, false>::integrate' requested here
~PhaseFieldDisplacementSpectralSplit()
^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:29:19: error: lambda capture 'emin' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
const auto df = [&emin](const real x)
                  ^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:115:1: note: in instantiation of member function 'tfel::material::PhaseFieldDisplacementSpectralSplit<tfel::material::ModellingHypothesis::GENERALISEDPLANESTRAIN, double, false>::integrate' requested here
~PhaseFieldDisplacementSpectralSplit()
^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:29:19: error: lambda capture 'emin' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
const auto df = [&emin](const real x)
                  ^
/builds/manta-project/manta/examples/phasefield/kalthoff/PhaseFieldDisplacementSpectralSplit.mfront:115:1: note: in instantiation of member function 'tfel::material::PhaseFieldDisplacementSpectralSplit<tfel::material::ModellingHypothesis::TRIDIMENSIONAL, double, false>::integrate' requested here
~PhaseFieldDisplacementSpectralSplit()
```

**Comment by thelfer** [comment]
@MorganeSteins Thanks for reporting.

For 2), just modify your `MFront` file to avoid the capture of `emin`. Beware however that this was required by some (buggy, i.e. not completly standard compliant) version of `gcc`.

For 1), would you tell which compiler and compiler flags you are using. I don't detect this issue.

**Comment by thelfer** [comment]
@MorganeSteins Don't get any issue with clang 14.0.0.

```bash
$ rm -fr include/ src/ && CXX=clang++ CXXFLAGS=`tfel-config --oflags --warning  --compiler-flags --includes` mfront --obuild --interface=generic SaintVenantKirchhoffElasticity.mfront 
Treating target : all
The following library has been built :
- libBehaviour.so :  SaintVenantKirchhoffElasticity_AxisymmetricalGeneralisedPlaneStrain SaintVenantKirchhoffElasticity_Axisymmetrical SaintVenantKirchhoffElasticity_PlaneStrain SaintVenantKirchhoffElasticity_GeneralisedPlaneStrain SaintVenantKirchhoffElasticity_Tridimensional
```

`tfel-config --oflags --warning` is very stringent:

```bash
$ tfel-config --warning
-Wlong-long -Wsign-promo -Wsign-conversion -Wzero-as-null-pointer-constant -Wmissing-declarations -Wswitch -Wsign-compare -pedantic -Wextra -Wsuggest-override -Wint-in-bool-context -Wregister -Wmisleading-indentation -Wnull-dereference -Wtautological-compare -Wshift-overflow -Wshift-negative-value -Wsizeof-array-argument -Wlogical-not-parentheses -Wswitch-bool -Wsequence-point -Wignored-qualifiers -Wold-style-cast -Wmissing-include-dirs -Winit-self -Wdouble-promotion -Wno-conversion -Wreorder -Wundef -Wunknown-pragmas -Wredundant-decls -Wpacked -Wno-deprecated-declarations -Wno-multichar -Wmissing-format-attribute -Wno-endif-labels -Wfloat-equal -Wreturn-type -Woverloaded-virtual -Wnon-virtual-dtor -Wctor-dtor-privacy -Wwrite-strings -Wcast-align -Wcast-qual -Wpointer-arith -Wshadow -W -Wall -Wno-c++98-compat -Wno-inconsistent-missing-destructor-override -Wmicrosoft -Wcomma -Wno-missing-variable-declarations -Winfinite-recursion -Wmove -Wrange-loop-analysis -Wno-missing-braces -Wno-global-constructors -Wno-exit-time-destructors -Wno-documentation-unknown-command -Wno-documentation -Wno-padded -Wno-c++98-compat-pedantic -Weverything
``` 

So maybe the warnings are related to an old version of TFEL ?

**Comment by MorganeSteins** [comment]
We are currently using clang-13 to compile version `tfel@4.0.0` which looks to be the last version of the library. I will see with the MANTA team if we can run the tests with clang-14 and see if warnings disappear. 
Thank you for checking this out !

**Comment by thelfer** [comment]
Hi @MorganeSteins. Looks like its been corrected in the `master` branch. I can reproduce the issue with `rliv-4.0`.

**Comment by thelfer** [comment]
Fixed in `rliv-4.0`. I strongly recommend that development versions of `MANTA` follows this branch (no major updates, only bug fixes).
